include/seqan3/range/view/get.hpp

```diff
@@ -64,7 +64,12 @@ inline auto const get = view::transform([] (auto && in) -> decltype(auto)
     using seqan3::get;
     static_assert(tuple_like_concept<decltype(in)>,
                   "You may only pass ranges to view::get whose reference_t models the tuple_like_concept.");
-    return get<index>(std::forward<decltype(in)>(in));
+
+    // we need to explicitly remove && around temporaries to return values as values (and not as rvalue references)
+    // we cannot simply cast to std::tuple_element_t (or set that as return value), because some tuples, like
+    // our cartesian_composition alphabets do not return that type when get is called on them (they return a proxy)
+    using ret_type = remove_rvalue_reference_t<decltype(get<index>(std::forward<decltype(in)>(in)))>;
```

Comment:
```suggestion
    using r_type = std::conditional_t<std::is_rvalue_reference_v<decltype(get<index>(std::forward<decltype(in)>(in)))>,
                                      std::remove_reference_t<decltype(get<index>(std::forward<decltype(in)>(in)))>,
                                      decltype(get<index>(std::forward<decltype(in)>(in)))>;
```

Reply:
This was my solution but I couldn't grasp yesterday why perfect forwarding using glvalues did not work. Now, I think I got it. The inner tuple was moved into the outer one but it was returned as xvalue, which is a dangling reference after the lambda finished I suppose? 
In general, I do not see a general purpose use case for `remove_rvalue_referene_t`?

Reply:
> In general, I do not see a general purpose use case for `remove_rvalue_referene_t`?

There were other situations were I defined two overloads where I could now use only one, e.g. the two overloads of assign_rank/assign_char free function: https://github.com/seqan/seqan3/blob/master/include/seqan3/alphabet/detail/member_exposure.hpp#L113

> This was my solution but I couldn't grasp yesterday why perfect forwarding using glvalues did not work. Now, I think I got it. The inner tuple was moved into the outer one but it was returned as xvalue, which is a dangling reference after the lambda finished I suppose?

Something like that.  The reference type of the nested zip:
```cpp
// originally:
tuple<tuple<int &, int &> &&, int &> &&
// now:
tuple<tuple<int &, int &>, int &>
```
The problem here is with the inner tuple: `&&` (and `const &` for that matter) extend the life-time of temporaries only until until their own lifetime ends. If you declare another reference pointing to the first, this doesn't extend the lifetime further. But this happens already when storing the reference (in our case inner tuple) in the (outer) tuple, that's why it's never safe to use `&&` (or owning `const &`)  as tuple element type.

I have added this small example to document this for someone else stumbling on the code comment or issue later:
```cpp
#include <tuple>

std::tuple<int const&> foobar()
{
        return {int{7}};
}

int main()
{
        auto t = foobar();
        return std::get<0>(t);
}
```
The return value is random on linux, always 255 on FreeBSD for some reason. Only address-sanatizer in release mode detects this as "stack-after-scope".

Reply:
As a matter of fact the reference type of the zip is: 
```cpp
// originally:
tuple<tuple<int &, int &>, int &>
```
But the lambda captures this as: `tuple<tuple<int &, int &>, int &> &&`  and the `get` will forward `tuple<int &, int &> &&` which causes the problem. And yes it is because the lifetime get's extended while it shouldn't. 👍 
But ok, if we need the remove_rvalue_reference in other places keep it, but perhaps instead of the overloads you could use the conditional statement to reduce boiler plate code a little?

Reply:
done